paperqa/llms.py

```diff
@@ -306,41 +326,50 @@ def validate_client(self):
             )
             raise ImportError(msg)
 
-        if self.client and not isinstance(self.client, QdrantClient):
+        if self.client and not isinstance(self.client, AsyncQdrantClient):
             raise TypeError(
-                f"'client' should be an instance of `qdrant_client.QdrantClient`. Got `{type(self.client)}`"
+                f"'client' should be an instance of AsyncQdrantClient. Got `{type(self.client)}`"
             )
 
         if not self.client:
             # Defaults to the Python based in-memory implementation.
-            self.client = QdrantClient(location=":memory:")
+            self.client = AsyncQdrantClient(location=":memory:")
 
         return self
 
+    async def _collection_exists(self) -> bool:
+        return await self.client.collection_exists(self.collection_name)
+
+    @override
     def clear(self) -> None:
-        super().clear()
+        """Synchronous clear method that matches parent class."""
+        loop = asyncio.get_event_loop()
+        if loop.is_running():
+            raise RuntimeError(
+                "Cannot call synchronous clear() from an async context. Use aclear() instead."
+            )
```

Comment:
So  since `Docs.clear_docs` is synchronous only, people interfacing with the base class `VectorStore` will need the sync implementation.

Do you think this will be a problem at all?